scripts/bravo_csv.py

```diff
@@ -286,6 +286,10 @@ def dilution(currentStep):
     # Read in the presets of minimum required conc for setup workset, maximum conc for dilution and minimum volume for dilution
     if "SMARTer Pico RNA" in currentStep.input_output_maps[0][0]['uri'].workflow_stages[0].workflow.name:
         preset = Dilution_preset['Smarter pico']
+    # Use the values set in the step UDFs; if not set, use the default values
+    min_required_conc = currentStep.udf['Minimum required conc for workset (ng/ul)'] if currentStep.udf['Minimum required conc for workset (ng/ul)'] else preset[0]
+    max_conc_for_dilution = currentStep.udf['Maximum conc for dilution (ng/ul)'] if currentStep.udf['Maximum conc for dilution (ng/ul)'] else preset[1]
+    min_vol_for_dilution = currentStep.udf['Minimum volume for dilution (ul)'] if currentStep.udf['Minimum volume for dilution (ul)'] else preset[2]
```

Comment:
Does this work if the property/key is not set? I'm not sure how the object that is `currentStep` works but for regular dictionaries you would get a KeyError like that:

```python
>>> d = { "a": 1, "b": 2 }
>>> x = d["c"] if d["c"] else 3
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
KeyError: 'c'
```

Also, do you want to allow values to be 0? Because now you don't and maybe that isn't obvious.

```python
>>> d = { "a": 0, "b": 2 }
>>> y = d["a"] if d["a"] else 99
>>> y
99
```

Reply:
Hi @jgruselius, if any of the step UDFs is not set, the key is still there but the value is empty. Thus there is no problem with the if else logics. And in LIMS the value ranges has been defined (no 0, negative or extremely large values).